lib/cog_api/http/relay_groups.ex

```diff
@@ -38,30 +38,19 @@ defmodule CogApi.HTTP.RelayGroups do
     |> ApiResponse.format_delete("The relay group could not be deleted")
   end
 
-  def update_memberships(action, %{name: name}, %{relay: relay_name}, %Endpoint{}=endpoint) do
-    case get_group_relay(name, relay_name, endpoint) do
-      {:ok, relay, relay_group} ->
-        update_membership(relay_group.id, relay.id, action, endpoint)
-      error ->
-        error
+  def update_memberships(action, %{name: name}, %{relays: relay_names}, endpoint) when is_list(relay_names) do
+    with {:ok, relay_group} <- show(%{name: name}, endpoint),
+         {:ok, relay_ids}   <- get_relay_ids(relay_names, endpoint) do
+      update_memberships(action, relay_group.id, relay_ids, endpoint)
     end
   end
-  def update_memberships(action, relay_group_id, relay_id, %Endpoint{}=endpoint) do
-    update_membership(relay_group_id, relay_id, action, endpoint)
-  end
-
-  defp get_group_relay(name, relay_name, endpoint) do
-    with {:ok, relay} <- Base.get_by(endpoint, "relays", name: relay_name)
-        |> ApiResponse.format(%{"relay" => CogApi.Resources.Relay.format}),
-      {:ok, relay_group} <- show(%{name: name}, endpoint),
-      do: {:ok, relay, relay_group}
-  end
-
-  defp update_membership(relay_group_id, relay_id, action, endpoint) do
+  def update_memberships(action, relay_group_id, relay_ids, endpoint) when is_list(relay_ids) do
     path = "relay_groups/#{relay_group_id}/relays"
-    Base.post(endpoint, path, %{relays: %{action =>  [relay_id]}})
+    Base.post(endpoint, path, %{relays: %{action => relay_ids}})
     |> ApiResponse.format(%{"relay_group" => RelayGroup.format})
   end
+  def update_memberships(action, relay_group_id, relay_id, endpoint) when is_binary(relay_id),
```

Comment:
You can also eliminate this function head and the guards (as well as any others that are similar) by simply wrapping `relay_ids` with `List.wrap` before you do anything else.

In that case, I'd also rename `relay_ids` to `relay_id_or_ids` to be a little more clear, but only if calling with a single ID is still actually something that can occur.